libpay-braintree/src/main/scala/com/wix/pay/braintree/BraintreeMerchantParser.scala

```diff
@@ -1,6 +1,6 @@
 package com.wix.pay.braintree
 
 trait BraintreeMerchantParser {

```

Comment:
How about making this type parameterized, like :
trait BraintreeMerchantParser[+T <: BraintreeMarchant] { 
   def parse(merchantKey: String): T
  def stringify(merchant: BraintreeMerchant): String
}
(this assuming you accept my previous comment, regarding renaming BraintreeMerchantBase to BraintreeMerchant)